analyzers/src/SonarAnalyzer.CSharp/Rules/ClearTextProtocolsAreSensitive.cs

```diff
@@ -0,0 +1,167 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2020 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.SyntaxTrackers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class ClearTextProtocolsAreSensitive : HotspotDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S5332";
+        private const string MessageFormat = "Using {0} protocol is insecure. Use {1} instead.";
+        private const string EnableSslMessage = "EnableSsl should be set to true.";
+        private const string HttpPattern = @"^http:\/\/(?!localhost|127.0.0.1).+";
+        private const string FtpPattern = @"^ftp:\/\/.*@(?!localhost|127.0.0.1)";
+        private const string Telnet = "telnet";
+        private const string TelnetPattern = @"^telnet:\/\/.*@(?!localhost|127.0.0.1)";
+        private const string TelnetPatternForIdentifier = @"(Telnet)(?![a-z])";
```

Comment:
Parenthesis should not be needed
```suggestion
        private const string TelnetPatternForIdentifier = @"Telnet(?![a-z])";
```